root/static/scripts/edit/URLCleanup.js

```diff
@@ -4528,16 +4528,19 @@ const CLEANUPS: CleanupEntries = {
     match: [new RegExp('^(https?://)?(www\\.)?operabase\\.com', 'i')],
     restrict: [LINK_TYPES.otherdatabases],
     clean: function (url) {
-      return url.replace(/^(?:https?:\/\/)?(?:www\.)?operabase\.com\/(artists|venues\/[\w-]+|works)\/(?:[^0-9]+)?([0-9]+).*$/, 'https://operabase.com/$1/$2');
+      url = url.replace(/^(?:https?:\/\/)?(?:www\.)?operabase\.com\//, 'https://operabase.com/');
+      url = url.replace(/^https:\/\/operabase\.com\/(venues\/[\w-]+|works)\/(?:[^0-9]+)?([0-9]+).*$/, 'https://operabase.com/$1/$2');
+      url = url.replace(/^https:\/\/operabase\.com\/(?:artists\/(?:[^0-9]+)?|[\w-]+a)([0-9]+).*$/, 'https://operabase.com/a$1');
+      return url;
     },
     validate: function (url, id) {
-      const m = /^https:\/\/operabase\.com\/(artists|venues|works)\//.exec(url);
+      const m = /^https:\/\/operabase\.com\/(a[0-9]+|venues|works)\/?/.exec(url);
```

Comment:
This is capturing the ID rather than just the prefix. To also validate the prefix for all prefixes:

```suggestion
      const m = /^https:\/\/operabase\.com\/(a|venues\/|works\/)[0-9]+$/.exec(url);
```

The below conditions would have to be adjusted accordingly.

Reply:
Well, `venues` is not that simple, this would need to be:
`const m = /^https:\/\/operabase\.com\/(a|venues\/[\w-]+\/|works\/)[0-9]+$/.exec(url);`

And then the check:
`result: /^venues\/[\w-]+\/$/.test(prefix),`

That said, I agree that does seem better as far as checks go, so if you're happy with that I'll push it.

Reply:
Then maybe something like that?

```suggestion
      const m = /^https:\/\/operabase\.com\/(?:(a)|(venues)\/[\w-]+\/|(works)\/)[0-9]+$/.exec(url);
```

And defining the prefix as the union:
```js
const prefix = m[1] || m[2] || m[3];
```